tests/src/test/scala/com/nvidia/spark/rapids/timezone/CpuTimeZoneDB.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.timezone
+
+import java.time.ZoneId
+
+import ai.rapids.cudf.{ColumnVector, DType, HostColumnVector}
+import com.nvidia.spark.rapids.Arm.withResource
+
+import org.apache.spark.sql.catalyst.util.DateTimeUtils
+import org.apache.spark.sql.catalyst.util.DateTimeUtils.daysToLocalDate
+
+object CpuTimeZoneDB {
+
+  def cacheDatabase(): Unit = {}
+
+  /**
+   * convert date or timestamp vector in a time zone to UTC timestamp
+   *
+   * @return
+   */
+  def convertToUTC(inputVector: ColumnVector, currentTimeZone: ZoneId): ColumnVector = {
```

Comment:
I am kind of confused by some of the code here.  I think the equivalent of these functions are `DateTimeUtils.fromUTCTime` and `DateTimeUtils.toUTCTime`.

https://github.com/apache/spark/blob/94607dd001b133a25dc9865f25b3f9e7f5a5daa3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala#L491-L505

Which ends up being implemented with `SparkDateTimeUtils.convertTz`

https://github.com/apache/spark/blob/94607dd001b133a25dc9865f25b3f9e7f5a5daa3/sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/SparkDateTimeUtils.scala#L130-L133

And for conversion to/from dates it is `SparkDateTimeUtils.daysToMicros` and `SparkDateTimeUtils.microsToDays`

https://github.com/apache/spark/blob/94607dd001b133a25dc9865f25b3f9e7f5a5daa3/sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/SparkDateTimeUtils.scala#L159-L172

To me if this is just for testing and getting things started it feels very much like we should just be using these methods instead of trying to implement something ourselves.  It also might be nice to split up the timestamp and date APIs so it is more clear what we expect to be passed in.

Reply:
Done.